src/Smalot/PdfParser/PDFObject.php

```diff
@@ -186,41 +201,206 @@ public function cleanContent(string $content, string $char = 'X')
         return $content;
     }
 
+    /**
+     * Takes a string of PDF document stream text and formats
+     * it into a multi-line string with one PDF command on each line,
+     * separated by \r\n. If the given string is null, or binary data
+     * is detected instead of a document stream then return an empty
+     * string.
+     *
+     * @internal
+     */
+    public function formatContent(?string $content): string
```

Comment:
Why don't you define it as `private` instead? It'll avoid adding the `@internal`

Reply:
Well, it's called explicitly in **PDFObjectTest.php**, but I can work around that with `ReflectionMethod`. Initially I thought it would be useful to be able to use this method publicly to format any old document stream, but that's probably not necessary.

Reply:
Maybe this can be applied to other `@internal` method?

Reply:
The less public methods we have the better, because `@internal` is just a label and can't be enforced.

_About the testing of private methods:_ I support the view that private functionality is not to test, because it is the objects obligation. One should only test public methods. In practice this might lead to complicated situations in which it is hard to cover a certain case sometimes.

> Initially I thought it would be useful to be able to use this method publicly to format any old document stream, but that's probably not necessary.

I was thinking that a method like that should be extracted from PDFObject and moved to a utility class or something. It is useful and might be handy outside of PDFObject context. But we should finalize this one first and then see. Because it is private now, we could extract it from PDFObject and make it available later on, if we want.

Reply:
> I was thinking that a method like that should be extracted from PDFObject and moved to a utility class or something. It is useful and might be handy outside of PDFObject context.

Definitely. For instance, I feel like my added function `parseDictionary()` duplicates much of the `protected` parsing functionality from **Parser.php** that breaks apart PDF 'dictionary' structures, which aren't only used in document streams, but trailer info, *etc*. In the future, there should probably be one global class or function that parses dictionaries (a fundamental PDF data structure) for all situations.